main.go

```diff
@@ -13,8 +14,11 @@ const PORT = ":8080"
 func main() {
 	models.InitDB("admin:admin@tcp(localhost:3308)/restaurantlisting?parseTime=true")
 
+	middlewarechain := middlewares.CreateMiddleWares(middlewares.Log, middlewares.SetHeader)
+
 	// Route Handlers / Endpoints
-	http.HandleFunc("/api/restaurants", controllers.Index)
+	http.HandleFunc("/api/restaurants", middlewarechain.Run(controllers.Index))
+	http.HandleFunc("/api/restaurants/", middlewarechain.Run(controllers.GetRestaurant))
```

Comment:
I changed this route from `/restaurant/` to `/restaurants/`, which is more conventional